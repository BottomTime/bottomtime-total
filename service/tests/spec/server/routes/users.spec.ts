import createTests from './users.create';
import retrieveTests from './users.retrieve';
import updateTests from './users.update';

describe('Users Routes', () => {
  describe('Creating Users', createTests);
  describe('Retrieving and Searching Users', retrieveTests);
  describe('Updating Account Properties', updateTests);
  it.todo(
    'Write email client so you can test password resets and email verifications',
  );
});
