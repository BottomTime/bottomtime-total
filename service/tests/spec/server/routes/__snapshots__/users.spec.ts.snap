// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Routes Creating Users Will create a new user based on the criteria 1`] = `"Welcome to Bottom Time!"`;

exports[`Users Routes Creating Users Will create a new user based on the criteria 2`] = `
"<!DOCTYPE html><html><head><title>Welcome to Bottom Time!</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css"><script src="https://kit.fontawesome.com/0abec2fadd.js" crossorigin="anonymous"></script></head><body><div class="hero is-success"><div class="hero-body container"><h1 class="title">Welcome to Bottom Time!</h1></div></div><section class="section"><div class="container"><p class="content block">Thanks for signing up with us! We're pleased to meet you! Here are a few useful links to get
you started with your new account...</p><div class="block"><span class="text-icon is-size-5 is-info"><span class="icon is-large"><i class="fas fa-envelope"></i></span><span>Verify Your Email</span></span><p class="content ml-6">First of all, please click&nbsp;<a href="http://localhost:8080/verifyEmail?email=Telly.Monahan%40gmail.com&token=Ca1NjSesP7fOJZ0dKDpAkvif7dDnh4S2">here</a>&nbsp;to confirm your email address. If you did not sign up to use Bottom Time and you think you
are receiving this email in error, then please disregard this message.</p></div><div class="block"><span class="text-icon is-size-5 is-info"><span class="icon is-large"><i class="fas fa-user"></i></span><span>Setup Your Profile</span></span><p class="content ml-6">Next, take a minute to setup your user profile and set your preferences. Click&nbsp;<a href="http://localhost:8080/profile">here</a>&nbsp;to begin!</p></div><div class="block"><span class="text-icon is-size-5 is-info"><span class="icon is-large"><i class="fas fa-edit"></i></span><span>Log Your Dives!</span></span><p class="content ml-6">Finally, head over to&nbsp;<a href="http://localhost:8080/logs">here</a>&nbsp;to start filling out your logbook! (You can even upload your dives from other apps!)</p></div><p class="content block">Cheers,<br>The BottomTime Team</p></div></section><footer class="footer"><div class="container has-text-centered is-size-7"><p><em>Copyright &copy; Chris Carleton, 2023</em></p><p><a href="http://localhost:8080/">Home Page</a> | <a href="mailto:admin@bottomti.me">admin@bottomti.me</a></p></div></footer></body></html>"
`;

exports[`Users Routes Email Verification and Password Resets Email Verification Will send a verification token to the user on demand 1`] = `"Verify your email address"`;

exports[`Users Routes Email Verification and Password Resets Email Verification Will send a verification token to the user on demand 2`] = `
"<!DOCTYPE html><html><head><title>Verify Your Email Address</title><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css"><script src="https://kit.fontawesome.com/0abec2fadd.js" crossorigin="anonymous"></script></head><body><div class="hero is-success"><div class="hero-body container"><h1 class="title">Verify Your Email Address</h1></div></div><section class="section"><div class="container"><p class="content block">Hi there Jimmy F.,</p><p class="content block">You are receiving this email because you requested a link to verify your email address with
the <a href="http://localhost:8080/">Bottom Time</a> application for social scuba diving! Click the link below
to complete the request!</p><div class="block box has-text-centered"><a href="http://localhost:8080/verifyEmail?email=jake123%40company.org&token=secrettoken">http://localhost:8080/verifyEmail?email=jake123%40company.org&amp;token=secrettoken</a></div><p class="content block">Cheers,<br>The Bottom Time Team!</p><p class="content block help">If you did not request this email and suspect something fishy is up then you can report this email
to <a href="mailto:admin@bottomti.me">admin@bottomti.me</a>.</p></div></section><footer class="footer"><div class="container has-text-centered is-size-7"><p><em>Copyright &copy; Chris Carleton, 2023</em></p><p><a href="http://localhost:8080/">Home Page</a> | <a href="mailto:admin@bottomti.me">admin@bottomti.me</a></p></div></footer></body></html>"
`;
